watcher.go

```diff
@@ -420,9 +419,15 @@ func (tp trackedPair) markCacheAccessed() trackedPair {
 	return tp
 }
 
+// IDer an interface that supports and ID
+type IDer interface {
+	ID() string
+}
```

Comment:
Interesting pattern! Noting this down to use some day in the future!

Reply:
The interface "-er" suffix has always been amusing to me

Reply:
I've been trying to be stricter with my inputs by defining more specific Interface types as needed. I feel there is probably a better point of balance that is more specific/detailed than I've done in the past.

And while I've just accepted the "-er" naming... it comes at a cost (eg. Watcherer, Renderer... :grimacing:)